### PR TITLE
Ye ehf 8237 resourceswelove

### PR DESCRIPTION
**Pull request by yasodaesarapu**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,86 +1,148 @@
...
+  }
+
+  state = {
+    isSeeAll: true,
+    allResourceLinks: [],
+    minResourceLinks: [],
+    mobileResourceLinks: [],
+    isMobileDevice: false,
+    resourceLinkDisplay: this.props.minResourceDisplay
   };
```
Comment (mfletcher-EH) [inline src/index.js]: Setting state from props is supposed to be bad, but finding a better way left me confused the last time I had lint complain. Moving things to other places in the lifecycle also brought up warnings. This is initial commit, and I have no good advice, so something to look into later.

File: src/component.scss [hunk 2]
```diff
@@ -1,28 +1,72 @@
...
+		color: color($ui-colors, black);
+	}
+
+	&__content {
 		display: block;
-		padding: 0 11px;
-		font-family: OpenSans, sans-serif;
+		
+		@media screen and (min-width: 768px) {
+			display: inline-table;
```
Comment (mfletcher-EH) [inline src/component.scss]: As discussed, there's no current breakpoint for 768px in style-helpers. Let's look into that.

File: src/index.js [hunk 3]
```diff
@@ -1,86 +1,148 @@
...
-    } = this.props;
+  renderShowCTA = () => {
+    const { allResourceLinks } = this.state;
+    return (
+      <div 
+        className="resources-we-love__cta"
+        onClick={() => this.toggleCTA(false,allResourceLinks)}
+      >
+        <a>
+          See All <Arrow direction={"right"} color={"#1b9f81"} size={10} strokeWidth={10} />
```
Comment (mfletcher-EH) [inline src/index.js]: Can we get this color from style-helpers?

File: src/index.js [hunk 4]
```diff
@@ -1,86 +1,148 @@
...
+      })
+    )
+  }
 
-    this.state = {
-      count: 0,
-      isActive: false
-    };
+  renderMobileResources = () => {
+    let currentResLinks = this.state.resourceLinkDisplay;
```
Comment (MiSTERSPOONS) [inline src/index.js]: After initial merge, we can refactor to update `this.state.resourceLinkDisplay;` based on whether the user clicks on `Show More` or `Show Less` and increment/decrement the iteration up to the `this.state.resourceLinkDisplay`

File: src/index.js [hunk 5]
```diff
@@ -1,86 +1,148 @@
...
-      buttonColor
-    } = this.props;
-
-    const { count } = this.state;
-
-    const buttonStyle = {
-      backgroundColor: buttonColor
-    };
-
+    const heartDuo = 'http://images.agoramedia.com/everydayhealth/gcms/heart-duo.png';
```
Comment (mfletcher-EH) [inline src/index.js]: Seems like this should be in a config somewhere vs. hard-coded in the middle of the file.

File: src/index.js (hunk 4 above)
Comment (yasodaesarapu) [reply src/index.js]: @MiSTERSPOONS Yes will refactor this part.

File: src/component.scss (hunk 2 above)
Comment (yasodaesarapu) [reply src/component.scss]: @mfletcher-EH Yes

File: src/index.js (hunk 3 above)
Comment (yasodaesarapu) [reply src/index.js]: @mfletcher-EH Yes updated.

File: src/index.js (hunk 5 above)
Comment (yasodaesarapu) [reply src/index.js]: @mfletcher-EH uploaded images to image server

File: src/index.js (hunk 1 above)
Comment (yasodaesarapu) [reply src/index.js]: @mfletcher-EH So basically it is a maximum resources to display for mobile if it has 7 or more. Basically these prop i am setting in default. May be i can just assign 7 here instead props? not sure that the correct way of doing . Please confirm

File: src/index.js [hunk 6]
```diff
@@ -1,86 +1,146 @@
...
-    containerClassName: PropTypes.string,
-    buttonColor: PropTypes.string
-  };
+class ResourcesWeLove extends React.PureComponent {
 
-  static defaultProps = {
-    containerClassName: '',
-    buttonColor: '#0000ff'
+  static propTypes = {
+    title: PropTypes.string,
```
Comment (clucaseh) [inline src/index.js]: Title,items, resourceItem, and ResourceContent are missing default prop definitions. They also should be a shape array is not allowed

File: src/index.js [hunk 7]
```diff
@@ -1,86 +1,146 @@
...
   }
 
-  render() {
-    const {
-      containerClassName,
-      buttonColor
-    } = this.props;
+  renderMobileResources = () => {
+    let currentResLinks = this.state.resourceLinkDisplay;
+    const { items } = this.props;
```
Comment (clucaseh) [inline src/index.js]: These methods. This one and above look unnecessary. Please refactor to use one method and use CSS where necessary to hide elements per the diff breakpoints

File: src/index.js [hunk 8]
```diff
@@ -1,86 +1,146 @@
...
+          </div>
+        )
+      }) 
+    ) 
+  }
 
-    const { count } = this.state;
+  renderCta = () => {
+    return (
+      this.state.isSeeAll ? this.renderShowCTA() : this.renderHideCTA()
```
Comment (clucaseh) [inline src/index.js]: Need to use destructuring here

File: src/index.js [hunk 9]
```diff
@@ -1,86 +1,146 @@
...
+    )
+  }
 
-    const buttonStyle = {
-      backgroundColor: buttonColor
-    };
+  renderShowCTA = () => {
+    const { allResourceLinks } = this.state;
```
Comment (clucaseh) [inline src/index.js]: Render show and hide should be one method. You can use args or props to achieve the different output

File: src/index.js [hunk 10]
```diff
@@ -1,86 +1,146 @@
...
-            {` ${count} `}
-          </span>
-          times
+  render() {
+    const heartDuo = 'https://cms.everydayhealth.com/images/default-source/react-component-images/heart-duo.png';
+    return (
+      <div className="resources-we-love">
+        <div className="resources-we-love__container">
+          <h2 className="resources-we-love__header">{this.props.title}</h2>
+            {this.isMobile ? this.state.mobileResourceLinks : this.renderDesktopResources()}
```
Comment (clucaseh) [inline src/index.js]: Destructuring should be done here too

File: src/index.js [hunk 11]
```diff
@@ -1,86 +1,146 @@
...
+    const heartDuo = 'https://cms.everydayhealth.com/images/default-source/react-component-images/heart-duo.png';
+    return (
+      <div className="resources-we-love">
+        <div className="resources-we-love__container">
+          <h2 className="resources-we-love__header">{this.props.title}</h2>
+            {this.isMobile ? this.state.mobileResourceLinks : this.renderDesktopResources()}          
         </div>
-
+        <div className="resources-we-love__heart-duo">
+          <img src={heartDuo} alt="heart-duo" />
```
Comment (clucaseh) [inline src/index.js]: Can we use SVG instead to avoid additional request

**Pull request merged**